docs/recipes/direct-audit/direct-audit.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const fs = require('fs');
+
+const makeDevtoolsLog = require('lighthouse/lighthouse-core/test/network-records-to-devtools-log.js');
+const ModuleDuplication = require('lighthouse/lighthouse-core/computed/module-duplication.js');
+const DuplicatedJavascript = require('lighthouse/lighthouse-core/audits/byte-efficiency/duplicated-javascript.js');
+const LegacyJavascript = require('lighthouse/lighthouse-core/audits/legacy-javascript.js');
+
+/**
+ * @param {Array<{url: string, content: string, map: any}>} scriptDatas
+ */
+function createGathererData(scriptDatas) {
+  const SourceMaps = scriptDatas.map(data => {
+    return {
+      scriptUrl: data.url,
+      map: data.map,
+    };
+  });
+  const ScriptElements = scriptDatas.map((data, i) => {
+    return {
+      requestId: `1000.${i}`,
+      src: data.url,
+      content: data.content,
+    };
+  });
+  const networkRecords = scriptDatas.map((data, i) => {
+    return {
+      requestId: `1000.${i}`,
+      url: data.url,
+      content: data.content,
+    };
+  });
+  networkRecords.push({url: 'https://www.example.com', resourceType: 'Document'});
+
+  const artifacts = {
+    URL: {finalUrl: 'https://www.example.com'},
+    devtoolsLogs: {defaultPass: makeDevtoolsLog(networkRecords)},
+    SourceMaps,
+    ScriptElements,
+  };
+
+  return {
+    artifacts,
+    networkRecords,
+  };
+}
+
+async function run() {
```

Comment:
let's allow @prateekbh to use this file as a lib.

i'm thinking we do a 

```js
module.exports = function analyze(arrayOfBundleAndMapCombos){ ...
```

so basically move L55-60 and L78 out into an `index.js` file or whatever.

Reply:
We discussed offline, but we do not want to have this file be a module that is used by some client. We don't want to support such an interface. This is just an example recipe.

Reply:
> We discussed offline, but we do not want to have this file be a module that is used by a client. We don't want to support such an interface. This is just an example recipe. Also it's too early to know what the requirements will be exactly :)

yeah, that makes sense, but maybe still consider leveraging `auditMode` in the recipe since absorbing the complexities of running audits is pretty much its purpose.

I don't know how gross it'll be to use this way, but possibly less than dealing with audit products directly :)

Reply:
@prateekbh connor convinced me its fine to keep this recipe super hardcoded because anyone we expect to use it is gonna have to do some work. :). 

but basically feel free to take the file and do the changes i suggested on your own.

if you continue using this and it makes sense for us to offer this as a real supported API, we can do that (eg. `lighthouse.analyzedJSBundles(arr)`), but it's too early right now to commit to that.

Reply:
Sounds good to me. Will get in touch